### PR TITLE
code review (thx @kaikreuzer) & discovery improvements

### PR DESCRIPTION
**Pull request by dimalo**
alright - one final chunk for 2.1 release

### CODE REVIEW

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiActorGatewayHandler.java [hunk 1]
```diff
@@ -29,19 +26,28 @@
...
 
     private static final int COLOR_TEMPERATURE_MAX = 6500;
     private static final int COLOR_TEMPERATURE_MIN = 1700;
 
-    private Integer lastBrightness;
+    private static final int DEFAULT_BRIGTHNESS_PCENT = 100;
+    private static final int DEFAULT_VOLUME_PCENT = 50;
+    private static final int DEFAULT_COLOR = 0xffffff;
+
+    private Integer lastBrigthness;
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiActorGatewayHandler.java:40]: typo: lastBrightness

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/XiaomiItemUpdateListener.java [hunk 2]
```diff
@@ -16,7 +16,22 @@
  * @author Patrick Boos - Initial contribution
  */
 public interface XiaomiItemUpdateListener {
+    /**
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/XiaomiItemUpdateListener.java:19]: add empty line

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/XiaomiItemUpdateListener.java [hunk 3]
```diff
@@ -16,7 +16,22 @@
...
+     * Callback method to notify the listener about a device state update
+     *
+     * @param sid the itemID of the device
+     * @param command the command type of the received message
+     * @param message the received message
+     *
+     * @author Patrick Boos - Initial contribution
+     */
+
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/XiaomiItemUpdateListener.java:28]: remove empty line

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/discovery/XiaomiBridgeDiscoveryService.java [hunk 4]
```diff
@@ -105,8 +111,20 @@ private void getGatewayInfo(JsonObject jobject) {
...
 
         ThingUID thingUID = new ThingUID(THING_TYPE_BRIDGE, serialNumber);
-        thingDiscovered(
-                DiscoveryResultBuilder.create(thingUID).withThingType(THING_TYPE_BRIDGE).withProperties(properties)
-                        .withLabel("Xiaomi Gateway").withRepresentationProperty(SERIAL_NUMBER).build());
+
+        Thing existing = discoveryServiceCallback.getExistingThing(thingUID);
+        if (existing != null) {
+            logger.debug("Bridge {} already exists - asking it for devices", thingUID);
+            // "Thing " + thingUID.toString() + " already exists"
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/discovery/XiaomiBridgeDiscoveryService.java:118]: remove

**Review by kaikreuzer — COMMENTED** [review]
Just a few very minor comments

**Review by kaikreuzer — APPROVED** [review]

**Comment by kaikreuzer** [comment]
@pboos To speed things up, I have already cherry-picked this commit to your master - the official PR is therefore already updated and you can simply close this PR here.

**Comment by pboos** [comment]
@kaikreuzer thank you :). didn't even know that is possible (that you can push to the pull request branch :D). Learning something new every day!
In that case closing this right now.

**Pull request closed without merging**